realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -481,7 +481,35 @@ public void downloadAllServerChanges() throws InterruptedException {
         // In Java we cannot lock on the Session object either since it will prevent any attempt at modifying the
         // lifecycle while it is in a waiting state. Thus we use a specialised mutex.
         synchronized (waitForChangesMutex) {
-            waitForChanges(DIRECTION_DOWNLOAD);
+            waitForChanges(DIRECTION_DOWNLOAD, Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    /**
+     * Calling this method will block until all known remote changes have been downloaded and applied to the Realm
+     * or the specified timeout was hit. This will involve network access, so calling this method should only be done
+     * from a non-UI thread.
+     * <p>
+     * This method cannot be called before the Realm has been opened.
+     *
+     * @throws IllegalStateException if called on the Android main thread.
+     * @throws InterruptedException if the timeout was hit or the thread was interrupted while downloading was in progress.
```

Comment:
"was hit": you mean "if the download took longer than the specified timeout setting".